File: src/Entity/Report.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use App\Repository\ReportRepository;

#[ORM\Entity(repositoryClass : ReportRepository::class)]
class Report {

    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id=null;

    #[ORM\Column(length : 250)]
    private ?string $type=null;

    #[ORM\Column(length : 500)]
    private ?string $description=null;

    #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(name: 'id_user', referencedColumnName: 'id')]
    private ?User $idUser=null;

    #[ORM\ManyToOne(targetEntity: Poste::class)]
    #[ORM\JoinColumn(name: 'id_poste', referencedColumnName: 'id')]
    private ?Poste $idPoste=null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getIdUser(): ?User
    {
        return $this->idUser;
    }

    public function setIdUser(?User $idUser): self
    {
        $this->idUser = $idUser;

        return $this;
    }

    public function getIdPoste(): ?Poste
    {
        return $this->idPoste;
    }

    public function setIdPoste(?Poste $idPoste): self
    {
        $this->idPoste = $idPoste;

        return $this;
    }
    public function getType() : ?string
    {
        return $this->type;
    }
    public function setType(?string $type) : self 
    {
        $this->type=$type;
        return $this;
    }
    public function getDescription() : ?string
    {
        return $this->description;
    }
    public function setDescription(?string $desc) : self 
    {
        $this->description=$desc;
        return $this;
    }

}

File: job4u-publication/job4u/src/com/job4u/entities/Report.java
package com.job4u.entities;

public class Report {

    private int id;
    private User user;
    private Poste poste;
    private String type;
    private String description;

    public Report() {
    }

    public Report(int id, User user, Poste poste, String type, String description) {
        this.id = id;
        this.user = user;
        this.poste = poste;
        this.type = type;
        this.description = description;
    }

    public Report(User user, Poste poste, String type, String description) {
        this.user = user;
        this.poste = poste;
        this.type = type;
        this.description = description;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Poste getPoste() {
        return poste;
    }

    public void setPoste(Poste poste) {
        this.poste = poste;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }


}